### ISSUE TITLE
Blockly.setTheme errors out if Blockly.getMainWorkspace() does not return a value

### ISSUE THREAD

**Issue by amber-cd** [google/blockly#2331]
### Problem statement

Blockly.setTheme errors out if Blockly.getMainWorkspace() does not return a value. (This can occur if using Blockly.setTheme with no workspace yet opened, such as when attempting to initialize Blockly information before injecting code.)

### Expected Behavior

If there is no main workspace yet,  Blockly.setTheme should just set Blockly.theme_ and move on.

### Actual Behavior

Blockly.setTheme errors out with "cannot read property getAllBlocks of undefined".

### Steps to Reproduce

Load up Blockly in any playground and attempt to call Blockly.setTheme before initializing the workspace.

### Operating System and Browser

* Desktop Chrome

(I expect it applies to all of them, though.)

### Additional Information

This is a less common use case, but certainly not an unthinkable one; I ran into it earlier today while adding Blockly.setTheme in order to update a nodejs Blockly project we've been using so as not to cause the blocks to error out when loaded up.

**Comment by RoboErikG** [comment]
Sounds like a reasonable thing to fix. =)

May not work for your setup, but you can also pass a theme object as part of the [inject parameters](https://developers.google.com/blockly/guides/configure/web/themes#set_your_theme).

**Comment by amber-cd** [comment]
Thanks! I'm afraid the injection parameters won't work for us here; our use case is a script which runs on Nodejs which can bulk update all of the generated JavaScript code in an app when we change the JavaScript generated by one or more blocks. We actually don't use inject itself anywhere -- just your basic

```
var workspace = new Blockly.Workspace();
Blockly.Xml.domToWorkspace(Blockly.Xml.textToDom(text), workspace);
Blockly.JavaScript.init(workspace);
Blockly.JavaScript.variableDB_.setVariableMap(workspace.getVariableMap());
return Blockly.Javascript.workspaceToCode(workspace);
```

deal, so it's entirely non-SVG workspaces here. I was actually really surprised to see that setStyle was an issue in non-SVG workspaces at all, but it looks like that's the case here.

For now I'm just manually setting `Blockly.theme_` if it hasn't been set yet in the code before generating the workspace, which works fine, but it always makes me a bit nervous to have to go outside of the API and directly reference private variables.

**Comment by RoboErikG** [comment]
Thanks for the use case. =) We're planning to fix it this week, so it should be resolved in the next merge to master.

**Comment by alschmiedt** [comment]
Hi Amber,

Thanks for bringing this up! 

Fix can be found here: #2335

**Comment by alschmiedt** [comment]
#2335
